dot/peerset/peerset.go

```diff
@@ -316,14 +332,15 @@ func (ps *PeerSet) updateTime() error {
 // be disconnected and a drop message for the peer is sent in order to disconnect.
 func (ps *PeerSet) reportPeer(change ReputationChange, peers ...peer.ID) error {
 	// we want reputations to be up-to-date before adjusting them.
-	if err := ps.updateTime(); err != nil {
-		return err
+	err := ps.updateTime()
+	if err != nil {
+		return fmt.Errorf("cannot update time: %w", err)
 	}
 
 	for _, pid := range peers {
 		rep, err := ps.peerState.addReputation(pid, change)
 		if err != nil {
-			return err
+			return fmt.Errorf("cannot add reputation (%s) to peer %s: %w", change, pid, err)
```

Comment:
nit the change and pid context should be added in `addReputation` instead of here